nnvm/python/nnvm/compiler/graph_util.py

```diff
@@ -146,3 +147,39 @@ def gradients(ys, xs, grad_ys=None):
         if isinstance(xs, list) else len(xs.list_output_names())
     ret = [grad_g.symbol[i] for i in range(nx)]
     return ret
+
+def split_last_op(graph):
```

Comment:
hopefully we have a formal graph splitting function in the future rather than this hacky function

Reply:
Yes. A general subgraph interface requires more thoughts and a more complete design. This is a short term solution for limited use cases.